cheatsheets/Protect_FileUpload_Against_Malicious_File.md

```diff
@@ -1,597 +1,138 @@
-                    }
-                }
-            }
-        }
-        catch (Exception e) {
-            safeState = false;
-            LOG.warn("Error during Excel file analysis !", e);
-        }
-        return safeState;
-    }
-
-}
-```
+#### Blacklist Extensions
 
-Detector for PowerPoint document:
-
-``` java
-import com.aspose.slides.IOleObjectFrame;
-import com.aspose.slides.IShape;
-import com.aspose.slides.ISlide;
-import com.aspose.slides.Presentation;
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
-
-import java.io.File;
-
-/**
- * Implementation of the detector for Microsoft PowerPoint document.
- *
- *
- */
-public class PowerPointDocumentDetectorImpl implements DocumentDetector {
-
-    /**
-     * LOGGER
-     */
-    private static final Logger LOG = LoggerFactory.getLogger(PowerPointDocumentDetectorImpl.class);
-
-    /**
-     * {@inheritDoc}
-     */
-    @SuppressWarnings("rawtypes")
-    @Override
-    public boolean isSafe(File f) {
-        boolean safeState = false;
-        try {
-            if ((f != null) && f.exists() && f.canRead()) {
-                // Load the file into the PowerPoint document parser
-                Presentation presentation = new Presentation(f.getAbsolutePath());
-                // First check on PowerPoint format skipped because:
-                // FileFormatInfo class is not provided for Aspose Slides API
-                // PresentationFactory.getInstance().getPresentationInfo() can be used 
-                // but the LoadFormat class miss format like POT or PPT XML
-                // Aspose API do not support PPT XML format
-                // Get safe state from presence of a VBA project in the presentation
-                safeState = (presentation.getVbaProject() == null);
-                // If presentation is safe then we pass to OLE objects analysis
-                if (safeState) {
-                    //Parse all slides of the presentation
-                    int totalOLEObjectCount = 0;
-                    for (ISlide slide : presentation.getSlides()) {
-                        for (IShape shape : slide.getShapes()) {
-                            //Check if the current shape is an OLE object
-                            if (shape instanceof IOleObjectFrame) {
-                                totalOLEObjectCount++;
-                            }
-                        }
-                    }
-                    // Update safe status flag according to number of OLE object found
-                    if (totalOLEObjectCount != 0) {
-                        safeState = false;
-                    }
-                }
-
-            }
-        } catch (Exception e) {
-            safeState = false;
-            LOG.warn("Error during PowerPoint file analysis !", e);
-        }
-        return safeState;
-    }
-}
-```
+> This should be only used when the [whitelist approach](#whitelist-extensions) fails, and is in no way a replacement for it.
```

Comment:
Needs big warnings

Reply:
Will add an emoji and maybe make them bold.

Reply:
I'd perhaps have it as something along the lines of.

"Blacklisting extensions is a bad idea and is very dangerous. Don't do it unless you have no other choice"